pyinteraph/path_analysis.py

```diff
@@ -0,0 +1,481 @@
+    """
+
+    # Initialize arrays of edges and their count
+    edges = np.array(graph.edges)
+    counts = np.array([graph[e[0]][e[1]]["weight"] for e in edges])
+    # Calculate percentage occurance for each path
+    perc = counts/counts.sum()
+    # Choose paths larger than the threshold
+    common_edges = edges[perc > threshold]
+    common_edges = list(map(tuple, common_edges))
+    return common_edges
+
+def get_metapath(graph, res_id, res_space, node_threshold, edge_threshold):
+    """Takes in a list of paths, an edge threshoold and a node threshold
+    and returns a list of metapaths where each metapath contains all 
+    nodes and edges above their respective thresholds.
+    """
+    
+    # Calculate all shortest paths
+    paths = get_all_shortest_paths(graph, res_id, res_space)
+    # Create graph from path list
+    paths_graph = get_graph_from_paths(paths)
+
+    # Get common nodes
+    common_nodes = get_common_nodes(paths, node_threshold)
+    # Get common edges
+    common_edges = get_common_edges(paths_graph, edge_threshold)
+
+    # Warn if no common nodes found
+    if len(common_nodes) == 0:
+        warn_str = "No Nodes found with a frequency higher than: {:f}"
+        log.warning(warn_str.format(node_threshold))
+    # Warn if no common edges found
+    if len(common_edges) == 0:
+        warn_str = "No Edges found with a frequency higher than: {:f}"
+        log.warning(warn_str.format(edge_threshold))
+
+    common_paths = []
+    # Common edges and nodes exist
+    if len(common_nodes) != 0 and len(common_edges) != 0:
+        # Find which paths have the common nodes and edges
+        for p in paths:
+            #print(p)
+            edges = [(p[i], p[i+1]) for i in range(len(p) - 1)]
+            # Check if required nodes are in path
+            for c_node in common_nodes:
+                if c_node in p:
+                    # Check if required edges are in path
+                    for c_edge in common_edges:
+                        # Add path if not already added
+                        if c_edge in edges and p not in common_paths:
+                            common_paths.append(p)
+        # Warn if no metapaths found
+        if len(common_paths) == 0:
+            warn_str = "No metapaths found."
+            log.warning(warn_str)
+    return common_paths, paths_graph
+
+def write_table(fname, table):
+    """Save sorted table as txt file. """
+    with open(f"{fname}.txt", "w") as f:
+        for p, s, t, l, sum_w, avg_w in table:
+            f.write(f"{p}\t{s}\t{t}\t{l}\t{sum_w}\t{avg_w}\n")
+
+
+def main():
+    ######################### ARGUMENT PARSER #########################
+    description = "Path analysis"
+    parser = argparse.ArgumentParser(description= description)
+
+    i_helpstr = ".dat file matrix"
+    parser.add_argument("-i", "--input-dat",
+                        dest = "input_matrix",
+                        help = i_helpstr,
+                        action = "store",
```

Comment:
If the action is simply `store`, you can avoid specifying it since it is the default.